src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -2134,9 +2134,20 @@ private void UpdateDatabase(ClusterOperationContext context, string type, Blitta
 
                     if (databaseRecordJson == null)
                     {
-                        if (updateCommand.ErrorOnDatabaseDoesNotExists)
-                            throw DatabaseDoesNotExistException.CreateWithMessage(databaseName, $"Could not execute update command of type '{type}'.");
-                        return;
+                        var shardIndex = valueNameLowered.AsSpan().LastIndexOf((byte)'$');
+                        if (shardIndex != -1) // try loading the sharded version, instead
```

Comment:
Because in this way, pretty much all our existing infrastructure behaves in the same manner. We just create "virtual" sharded dbs, but we have to specialize a lot less to support it.